packages/frontend/components/primitives/tooltip/component.tsx

```diff
@@ -0,0 +1,91 @@
+﻿import { useRef, useEffect, useState, useCallback } from 'react';
+import React from 'react';
+import * as ReactDOM from 'react-dom';
+import * as styles from './styles';
+import type { TooltipProps } from './types';
+
+export const Tooltip = ({ trigger, children }: TooltipProps) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const bodyRef = useRef<HTMLDivElement>();
+  const triggerWrapperRef = useRef<HTMLDivElement>();
+  let timer: NodeJS.Timeout | null = null;
+  const timeout = 500;
```

Comment:
pass as a prop with default value 500,
store timer in ref. const timerRef = useRef()